pkg/sysinfo/cgroup2_linux.go

```diff
@@ -65,11 +65,9 @@ func getSwapLimitV2() bool {
 		return false
 	}
 
-	cGroupPath := path.Join("/sys/fs/cgroup", g, "memory.swap.max")
-	if _, err = os.Stat(cGroupPath); os.IsNotExist(err) {
-		return false
-	}
-	return true
+	cGroupPath := path.Join(swap.cgMntPoint(), g, "memory.swap.max")
+	/* In case cGroupPath is the root cgroup, detect swap another way */
+	return exists(cGroupPath) || swap.enabled()
```

Comment:
I’m not convinced of this.
The function should return false when the kernel supports the swap but has no support for cgrouping the swap.